### PR TITLE
[DOCS] Expand the step that enables the remote cluster server

### PR DESCRIPTION
**Pull request by abdonpijpelink**
This PR expands the step where the remote cluster server is enabled, by splitting it into three sub-bullets for each of the `remote_cluster_server.enabled`, `remote_cluster.host`, and `remote_cluster.port` settings. 

Also adds a warning that for single node clusters, remote cluster server traffic may be bound to the local interface by default, preventing remote clusters from connecting.

### CODE REVIEW

File: docs/reference/modules/cluster/remote-clusters-api-key.asciidoc [hunk 1]
```diff
@@ -64,11 +64,18 @@ information, refer to https://www.elastic.co/subscriptions.
...
-
+. Enable the remote cluster server on every node of the remote cluster. In 
+`elasticsearch.yml`:
+.. Set <<remote-cluster-network-settings,`remote_cluster_server.enabled`>> to 
+`true`.
+.. Configure the bind and publish address for remote cluster server traffic, for
+example using <<remote-cluster-network-settings,`remote_cluster.host`>>.
+This defaults to the value of `transport.bind_host`, which in turn defaults to
+the value of `network.host`, which defaults to `_local_`. By default, for single
+node clusters, this means remote cluster traffic will be bound to the local
```
Comment (abdonpijpelink) [inline docs/reference/modules/cluster/remote-clusters-api-key.asciidoc:75]:
```suggestion
node clusters, this means remote cluster traffic is bound to the local 
```

File: docs/reference/modules/cluster/remote-clusters-api-key.asciidoc [hunk 2]
```diff
@@ -64,11 +64,18 @@ information, refer to https://www.elastic.co/subscriptions.
...
+. Enable the remote cluster server on every node of the remote cluster. In 
+`elasticsearch.yml`:
+.. Set <<remote-cluster-network-settings,`remote_cluster_server.enabled`>> to 
+`true`.
+.. Configure the bind and publish address for remote cluster server traffic, for
+example using <<remote-cluster-network-settings,`remote_cluster.host`>>.
+This defaults to the value of `transport.bind_host`, which in turn defaults to
+the value of `network.host`, which defaults to `_local_`. By default, for single
+node clusters, this means remote cluster traffic is bound to the local 
+interface and remote clusters can't connect.
```
Comment (ywangd) [inline docs/reference/modules/cluster/remote-clusters-api-key.asciidoc:76]: Two comments:
1. The full default chain has a few extra hops. It is `remote_cluster.host` -> `transport.bind_host` -> `transport.host` -> `network.bind_host` -> `network.host` -> `_local_`, where `->` means defaults to. Not sure if this detail matters. Please feel free to paraphrase.
2. This default chain is effective regardless of the discovery type. For example, you can form a multiple-nodes cluster on a single machine. All nodes will bind to `_local_` by default.

Comment (abdonpijpelink) [reply docs/reference/modules/cluster/remote-clusters-api-key.asciidoc]: I think you're right, it may be too much detail. How about rewording it as:

_Configure the bind and publish address for remote cluster server traffic, for example using [`remote_cluster.host`](http://localhost:8000/guide/modules-network.html#remote-cluster-network-settings). Without configuring the address, remote cluster traffic may be bound to the local interface, and as a result remote clusters can’t connect._

Comment (ywangd) [reply docs/reference/modules/cluster/remote-clusters-api-key.asciidoc]: Mostly good to me. Just wanna be a bit more specific on "remote clusters can't connect". If all clusters running directly on the local machine, they should be connect to each other. Not sure how much it matters. But that's how I usually test locally.

Comment (abdonpijpelink) [reply docs/reference/modules/cluster/remote-clusters-api-key.asciidoc]: Good point. I've changed it into: 

_Without configuring the address, remote cluster traffic may be bound to the local interface, and remote clusters running on other machines can't connect._

**Review by ywangd — APPROVED** [review]
LGTM
